### PR TITLE
feat: update docs, README, and add REST API

### PR DESCRIPTION
**Pull request by ServiceDog**

### CODE REVIEW

File: docs/sdd.md [hunk 1]
```diff
@@ -20,9 +20,9 @@ Status | Draft
 
 Document | Description
 --- | ---
-:construction: High-Level Concept of Operations (CONOPS) | Overview of Arrow microservices.
-:construction: High-Level Interface Control Document (ICD) | Interfaces and frameworks common to all Arrow microservices.
-:construction: Requirements - `svc-FIXME` | Requirements and user stories for this microservice.
+[High-Level Concept of Operations (CONOPS)](https://github.com/Arrow-air/se-services/blob/develop/docs/conops.md) | Overview of Arrow microservices.
+[High-Level Interface Control Document (ICD)](https://github.com/Arrow-air/se-services/blob/develop/docs/icd.md)  | Interfaces and frameworks common to all Arrow microservices.
+[Requirements - `svc-FIXME`](https://nocodb.arrowair.com/dashboard/#/nc/view/d1bb0a51-e22f-4b91-b1c5-66f11f4f861b) | Requirements and user stories for this microservice.
```
Comment (GoodluckH) [inline docs/sdd.md:25]: I think there's a way to generate a link for a specific svc. See svc-assets or pricing.

File: server/src/main.rs [hunk 2]
```diff
@@ -44,13 +166,48 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+    // --------------------------------------------------
+    // START REST SECTION
+    // This section should be removed if there is no REST interface
+    // --------------------------------------------------
+
+    // Allow option to only generate the spec file to a given location
+    let args = Cli::parse();
+    if let Some(target) = args.openapi {
+        return generate_openapi_spec(&target);
+    }
```
Comment (GoodluckH) [inline server/src/main.rs:201]: Maybe add some instruction in README on how to use the CLI? Especially for generating openapi locally.

File: server/src/main.rs [hunk 3]
```diff
@@ -1,14 +1,131 @@
...
+//!     <a href="https://github.com/Arrow-air/svc-template-rust/tree/main">
+//!         <img src="https://github.com/arrow-air/svc-template-rust/actions/workflows/rust_ci.yml/badge.svg?branch=main" alt="Rust Checks">
+//!     </a>
+//!     <a href="https://discord.com/invite/arrow">
+//!         <img src="https://img.shields.io/discord/853833144037277726?style=plastic" alt="Arrow DAO Discord">
+//!     </a>
+//!     <br><br>
+//! </div>
+//!
+//! `svc-template-rust` is responsible for FIXME
```
Comment (owlot) [inline server/src/main.rs:21]: I prefer using an include for the docs like I've done for svc-storage:
https://github.com/Arrow-air/svc-storage/blob/develop/server/src/lib.rs

File: client-rest/Cargo.toml
Comment (owlot) [inline client-rest/Cargo.toml:1]: Missing `[package]` info. Should reflect the same fields as `client-grpc/Cargo.toml` has but use the workspace provided information (should be updated for `client-grpc/Cargo.toml` as well):
```
categories.workspace   = true
edition.workspace      = true
homepage.workspace     = true
license.workspace      = true
license-file.workspace = true
repository.workspace   = true
```

File: server/Cargo.toml
Comment (owlot) [inline server/Cargo.toml:1]: With the addition of the `[workspace.package]` section in the `Cargo.toml` file of our root directory, you should now be able to use the following in this file:
```
categories.workspace   = true
edition.workspace      = true
homepage.workspace     = true
license.workspace      = true
license-file.workspace = true
repository.workspace   = true
```

Leaving only `description`, `keywords`, `name` and `version` to be filled out specifically.
Example: https://github.com/Arrow-air/svc-storage/blob/develop/server/Cargo.toml

Comment (ServiceDog) [reply server/Cargo.toml:1]: https://github.com/Arrow-air/svc-template-rust/pull/22/commits/fc42b589509567e5059fb64c73e66fb672e927b1

File: client-rest/Cargo.toml
Comment (ServiceDog) [reply client-rest/Cargo.toml:1]: https://github.com/Arrow-air/svc-template-rust/pull/22/commits/fc42b589509567e5059fb64c73e66fb672e927b1

File: docs/sdd.md (hunk 1 above)
Comment (ServiceDog) [reply docs/sdd.md]: Replaced with a FIXME to use a URL for your module's specific view: https://github.com/Arrow-air/svc-template-rust/pull/22/commits/fc42b589509567e5059fb64c73e66fb672e927b1

**Review by GoodluckH — APPROVED** [review]

**Pull request merged**